### ISSUE TITLE
too-few-public-methods reported on new @dataclass

### ISSUE THREAD

**Issue by MisterY** [PyCQA/pylint#1793]
### Steps to reproduce
1. 
```
from dataclasses import dataclass

@dataclass
class ScheduledTxSearchModel:
    date_from = None
    date_to = None
```

### Current behavior
too-few-public-methods reported

### Expected behavior
@dataclass should not have methods. Therefore, no R message should be triggered.

### pylint --version output

pylint 1.8.1,
astroid 1.6.0
Python 3.7.0a3 (v3.7.0a3:90a6785, Dec  5 2017, 22:04:17) [MSC v.1900 32 bit (Intel)]

**Comment by PCManticore** [comment]
Yes, we planned long ago to ditch this check, but didn't get to it just yet

**Comment by gpakosz** [comment]
The warning isn't properly disabled, try with:

```
from dataclasses import dataclass

@dataclass(frozen=True)
class Test:
    some_integer: int
```

```
$ pylint --version
pylint 2.1.1
astroid 2.0.4
Python 3.7.0 (default, Jul 15 2018, 10:44:58)
[GCC 8.1.1 20180531]
```

```
$ pylint ./test.py
************* Module test
test.py:1:0: C0111: Missing module docstring (missing-docstring)
test.py:3:0: C0111: Missing class docstring (missing-docstring)
test.py:3:0: R0903: Too few public methods (0/2) (too-few-public-methods)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```
